cmd/build/v2/services.go

```diff
@@ -43,7 +43,25 @@ func (bc *OktetoBuilder) GetServicesToBuild(ctx context.Context, manifest *model
 
 		svc := service
 		g.Go(func() error {
-			return bc.checkServicesToBuild(svc, manifest, toBuild)
+			digest, err := bc.getDigestFromService(svc, manifest)
+			if err != nil {
+				return err
+			}
+			if digest == "" {
+				toBuild <- svc
+				return nil
+			}
+			oktetoLog.Debug("Skipping build for image for service")
+
+			bc.SetServiceEnvVars(svc, digest)
+
+			if manifest.Deploy != nil && manifest.Deploy.ComposeSection != nil && manifest.Deploy.ComposeSection.Stack != nil {
+				stack := manifest.Deploy.ComposeSection.Stack
+				if svcInfo, ok := stack.Services[svc]; ok && svcInfo.Image == "" {
+					stack.Services[svc].Image = fmt.Sprintf("${OKTETO_BUILD_%s_IMAGE}", strings.ToUpper(strings.ReplaceAll(svc, "-", "_")))
```

Comment:
when is this image value from an stack loaded as env?

Reply:
We have to load the compose twice. The first one we need to get the builds from it, we add the build variables and then the second one we use it to load the env vars. We'll probably need to revisit this logic when we move to the official docker-compose serialiser

Reply:
I said that because I saw that for a service not declared as a stack I see https://github.com/okteto/okteto/blob/658767f05dba71c00109a322ecd81f2263c72722/cmd/build/v2/services.go#L56 which loads the necessary environment variables but for services declared as part of a stack I don't see this environment variable loading